src/graph/LookupExecutor.cpp

```diff
@@ -0,0 +1,555 @@
+        }
+        yieldClauseWrapper_ = std::make_unique<YieldClauseWrapper>(clause);
+        auto *varHolder = ectx()->variableHolder();
+        auto status = yieldClauseWrapper_->prepare(inputs_.get(), varHolder, yields_);
+        if (!status.ok()) {
+            return status;
+        }
+        for (auto *col : yields_) {
+            if (!col->getFunName().empty()) {
+                return Status::SyntaxError("Not supported yet");
+            }
+            auto* prop = col->expr();
+            if (prop->kind() != Expression::kAliasProp) {
+                return Status::SyntaxError("Expressions other than AliasProp are not supported");
+            }
+            auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
+            if (schema->getFieldIndex(aExpr->prop()->c_str()) < 0) {
+                LOG(ERROR) << "Unknown column " << aExpr->prop()->c_str();
+                return Status::Error("Unknown column `%s' in schema", aExpr->prop()->c_str());
+            }
+            returnCols_.emplace_back(aExpr->prop()->c_str());
+        }
+    }
+    return Status::OK();
+}
+
+Status LookupExecutor::optimize() {
+    Status status = Status::OK();
+    do {
+        status = checkFilter();
+        if (!status.ok()) {
+            break;
+        }
+        status = chooseIndex();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    if (!status.ok()) {
+        LOG(ERROR) << "Preparing failed: " << status;
+        return status;
+    }
+    return status;
+}
+
+Status LookupExecutor::traversalExpr(const Expression *expr) {
+    switch (expr->kind()) {
+        case nebula::Expression::kLogical : {
+            auto* lExpr = dynamic_cast<const LogicalExpression*>(expr);
+            if (lExpr->op() == LogicalExpression::Operator::XOR) {
+                return Status::SyntaxError("Syntax error : %s", lExpr->toString().c_str());
+            } else if (lExpr->op() == LogicalExpression::Operator::OR) {
+                skipOptimize_ = true;
+            }
+            auto* left = lExpr->left();
+            traversalExpr(left);
+            auto* right = lExpr->right();
+            traversalExpr(right);
+            break;
+        }
+        case nebula::Expression::kRelational : {
+            std::string prop;
+            VariantType v;
+            auto* rExpr = dynamic_cast<const RelationalExpression*>(expr);
+            auto* left = rExpr->left();
+            auto* right = rExpr->right();
+            /**
+             *  TODO (sky) : Support WHERE ：tag1.col2 != tag1.col3
+             *  Handler error for FuncExpr or  ArithmeticExpr contains
+             *  AliasPropExpr , for example :
+             *  WHERE lookup_tag_2.col2 > (lookup_tag_2.col3 - 100)
+             */
+            if (left->kind() == nebula::Expression::kAliasProp &&
+                right->kind() == nebula::Expression::kAliasProp) {
```

Comment:
In your case in comments,   the right expression  kind is not nebula::Expression::kAliasProp

Reply:
We'd better remove the if condition.